### PR TITLE
Task6

### PR DESCRIPTION
**Pull request by Vanad1um**

### CODE REVIEW

File: task6/data.js [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+
+    for (let i = 0; i < adList.length; i++) {
+        if (adList[i].id === id) {
+            indexByID = i;
+        }
+    }
+    if (typeof indexByID === 'undefined')
+        return false;
+
+    let tempObject = adList[indexByID];
```
Comment (YauheniyaBasaranovich) [inline task6/data.js:409]: Here it's better to copy object, not the link only

File: task6/data.js [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+console.log('getAd(\'1\')', getAd('1'));
+
+function removeAd(id) {
+    if (typeof id === 'string') {
+        for (let i = 0; i < adList.length; i++) {
+            if (adList[i].id === id) {
+                adList.splice(i, 1);
+                return true;
+            }
+        }
```
Comment (YauheniyaBasaranovich) [inline task6/data.js:453]: `Array.fidIndex`

File: task6/data.js [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+
+    if(adItem.id || adItem.vendor || adItem.createdAt){
+        return false;
+    }
+
+    for (let i = 0; i < adList.length; i++) {
+        if (adList[i].id === id) {
+            indexByID = i;
+        }
+    }
```
Comment (YauheniyaBasaranovich) [inline task6/data.js:405]: `Array.findIndex`

File: task6/data.js [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+console.log('getAds(0, 10, {vendor: \'Chair Service\'})', getAds(0, 10, {vendor: 'Chair Service'}));
+console.log('getAds(-1, 10)', getAds(-1, 10));
+console.log('getAds(0, 10, 16)', getAds(0, 10, 16));
+
+function getAd(id) {
+    if (typeof id === 'string') {
+        for (let i = 0; i < adList.length; i++) {
+            if (adList[i].id === id)
+                return adList[i];
+        }
```
Comment (YauheniyaBasaranovich) [inline task6/data.js:328]: `Array.find`

**Review by YauheniyaBasaranovich — APPROVED** [review]
Approved, but why did you remove UI folder?

**Comment by Vanad1um** [comment]
Well, now i do not understand what my every pr should contain. I thought that it should containall my folders that are not merged yet. Isn't that true?

**Comment by YauheniyaBasaranovich** [comment]
> Well, now i do not understand what my every pr should contain. I thought that it should containall my folders that are not merged yet. Isn't that true?

So, actually, you're writing a single project. Each task is a step towards your course goal -  fully working application. As you completed task3, there's no need to create new HTML and CSS files for each task - just use existing one. The same is applicable for all further tasks.

**Pull request merged**